data/tools/wesnoth/wmlparser3.py

```diff
@@ -371,11 +373,14 @@ def parse_binary(self, binary: bytes, defines="") -> RootNode:
         """
         Parse a chunk of binary WML.
         """
-        temp = tempfile.NamedTemporaryFile(prefix="wmlparser_",
-                                           suffix=".cfg")
+        td, tmpfilePath = tempfile.mkstemp(prefix="wmlparser_",
+                                            suffix=".cfg")
+        temp = open(tmpfilePath, 'wb')
```

Comment:
I would use "with open()" instead of open() and close().

Reply:
Ok pull request updated. Awaiting feedback.